service/controller/app/v1/resource/configmap/current.go

```diff
@@ -36,23 +39,36 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding configmap %#q in namespace %#q", name, r.chartNamespace))
 
-	chart, err := cc.K8sClient.CoreV1().ConfigMaps(r.chartNamespace).Get(name, metav1.GetOptions{})
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
+
+	configmap, err := cc.K8sClient.CoreV1().ConfigMaps(r.chartNamespace).Get(name, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
 		// Return early as configmap does not exist.
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find configmap %#q in namespace %#q", name, r.chartNamespace))
 		return nil, nil
 	} else if tenant.IsAPINotAvailable(err) {
-		// We should not hammer tenant API if it is not available, the tenant cluster
-		// might be initializing. We will retry on next reconciliation loop.
+		// We should not hammer tenant API if it is not available. We cancel
+		// the reconciliation because its likely following resources will also
+		// fail.
```

Comment:
The configmap resource is followed by the secret, chart and status resources that connect to tenant clusters. So we cancel the reconciliation since they are likely to also fail.